File: backend/src/main/java/com/e108/be/domain/dog/dto/response/DogProfileResponse.java
package com.e108.be.domain.dog.dto.response;

import com.e108.be.domain.dog.entity.Dog;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
public class DogProfileResponse {
    private Long dogId;
    private String name;
    private String breed;
    private LocalDate birthDate;
    private BigDecimal weight;
    private String gender;

    public DogProfileResponse(Dog dog) {
        this.dogId = dog.getId();
        this.name = dog.getName();
        this.breed = dog.getBreed();
        this.birthDate = dog.getBirthDate();
        this.weight = dog.getWeight();
        this.gender = dog.getGender();
    }
}

File: frontend/app/src/main/java/com/frontend/domain/model/DogProfileResponse.kt
package com.frontend.domain.model

data class DogProfileResponse(
    val dogId: Long,
    val name: String,
    val breed: String,
    val birthDate: String,   // "2020-03-08"
    val weight: Double,
    val gender: String       // "MALE" | "FEMALE"
)
